### PR TITLE
KNL-1550 - SQLService puts strings into preparedstatements as bytes

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

**Comment by buckett** [comment]
I'd noticed the lack of `String` support before and assumed it was for Oracle CLOB cases.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java [hunk 1]
```diff
@@ -2235,6 +2235,11 @@ else if (fields[i] instanceof Boolean)
 					pstmt.setBoolean(pos, ((Boolean) fields[i]).booleanValue());
 					pos++;
 				}
+				else if (fields[i] instanceof String)
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java]: Could be combined in the instanceof check above.

Comment (jonespm) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java]: I don't understand which one? The only other one I see involving string is a null check.

**Comment by jonespm** [comment]
Whatever you did to this does not compile

File: kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java [hunk 2]
```diff
@@ -2178,73 +2180,59 @@ public void ddl(ClassLoader loader, String resource)
...
-			{
-				if (fields[i] == null || (fields[i] instanceof String && ((String) fields[i]).length() == 0))
-				{
-					// treat a Java null as an SQL null,
-					// and ALSO treat a zero-length Java string as an SQL null
-					// This makes sure that Oracle vs MySQL use the same value
-					// for null.
+		if ((fields != null) && (fields.length > 0)) {
+			for (Object field : fields) {
+				if (field == null) {
```
Comment (buckett) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java]: Are you sure that we don't have any code that requires the empty string to `null` conversion?

**Comment by buckett** [comment]
Can we not mix brace style in the same file please.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java [hunk 3]
```diff
@@ -2178,73 +2180,59 @@ public void ddl(ClassLoader loader, String resource)
...
+					// Treat a Java null as an SQL null.
+					// This makes sure that Oracle vs MySQL use the same value for null.
 					sqlServiceSql.setNull(pstmt, pos);
 
 					pos++;
```
Comment (jonespm) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java]: You could probably move this pos++ incrementer to the bottom of the for loop now too. It seems like they could have justed use i+1 in the original case.

**Comment by ern** [comment]
Sorry that was my IDE!

**Pull request merged**